### PR TITLE
Add near and far clipping distance to ROS2 Camera sensor.

### LINKED ISSUE #417: Add near and far clipping distance to ROS2 Camera sensor
Add 'Near clip distance' and 'Far clip distance' to 'ROS2 Camera Sensor'.

### PR DESCRIPTION
**Pull request by michalpelka**
Fixes #417 .

### CODE REVIEW

File: Gems/ROS2/Code/Source/Camera/CameraSensorDescription.cpp [hunk 1]
```diff
@@ -37,6 +41,11 @@ namespace ROS2
         AZ_Assert(
             m_cameraConfiguration.m_width > 0 && m_cameraConfiguration.m_height > 0, "Camera resolution dimensions should be above zero");
         AZ_Assert(!m_cameraName.empty(), "Camera name cannot be empty");
+        AZ_Assert(m_cameraConfiguration.m_nearClipDistance > 0.0f, "Near clip distance should be above zero");
+        AZ_Assert(m_cameraConfiguration.m_farClipDistance > 0.0f, "Near clip distance should be above zero");
```
Comment (pijaro) [inline Gems/ROS2/Code/Source/Camera/CameraSensorDescription.cpp:45]:
```suggestion
        AZ_Assert(m_cameraConfiguration.m_farClipDistance > 0.0f, "Far clip distance should be greater than the near plane dsitance");
```

File: Gems/ROS2/Code/Source/Camera/CameraSensorDescription.cpp [hunk 2]
```diff
@@ -37,6 +41,11 @@ namespace ROS2
         AZ_Assert(
             m_cameraConfiguration.m_width > 0 && m_cameraConfiguration.m_height > 0, "Camera resolution dimensions should be above zero");
         AZ_Assert(!m_cameraName.empty(), "Camera name cannot be empty");
+        AZ_Assert(m_cameraConfiguration.m_nearClipDistance > 0.0f, "Near clip distance should be above zero");
```
Comment (jhanca-robotecai) [inline Gems/ROS2/Code/Source/Camera/CameraSensorDescription.cpp:44]:
```suggestion
        AZ_Assert(m_cameraConfiguration.m_nearClipDistance > 0.0f, "Near clip distance should be greater than zero");
```

File: Gems/ROS2/Code/Source/Camera/CameraSensorDescription.cpp [hunk 3]
```diff
@@ -37,6 +41,11 @@ namespace ROS2
...
+        AZ_Assert(m_cameraConfiguration.m_farClipDistance > 0.0f, "Far clip distance should be greater than the near plane dsitance");
```
Comment (jhanca-robotecai) [inline Gems/ROS2/Code/Source/Camera/CameraSensorDescription.cpp:45]:
```suggestion
        AZ_Assert(m_cameraConfiguration.m_farClipDistance > m_cameraConfiguration.m_nearClipDistance , "Far clip distance should be greater than the near plane distance");
```

**Review by jhanca-robotecai — APPROVED** [review]

File: Gems/ROS2/Code/Source/Camera/CameraSensorDescription.cpp [hunk 4]
```diff
@@ -37,6 +41,11 @@ namespace ROS2
         AZ_Assert(
             m_cameraConfiguration.m_width > 0 && m_cameraConfiguration.m_height > 0, "Camera resolution dimensions should be above zero");
         AZ_Assert(!m_cameraName.empty(), "Camera name cannot be empty");
+        AZ_Assert(m_cameraConfiguration.m_nearClipDistance > 0.0f, "Near clip distance should be greater than zero");
+        AZ_Assert(m_cameraConfiguration.m_farClipDistance > m_cameraConfiguration.m_nearClipDistance , "Far clip distance should be greater than the near plane distance");
```
Comment (adamdbrw) [inline Gems/ROS2/Code/Source/Camera/CameraSensorDescription.cpp:45]: These should probably be added to OnValidate, but it's ok

**Review by adamdbrw — APPROVED** [review]

**Review by pawelbudziszewski — APPROVED** [review]

**Pull request merged**